### DISCUSSION TITLE
A researcher's suggestions for enhancing Trilium

### DISCUSSION THREAD

**Discussion by Waxwood** [TriliumNext/Trilium#1511] [Ideas]
**1. Continue to improve the interface.**
The UI is already very good and better than that of Joplin which takes up far too much screen space, and requires the eye to travel here and there to identify the organizational structure. Trilium, on the other hand, allows a clear understanding of the data structure/organization with just a glance. 

The very best note-taking applications, MyInfo, MyBase, RightNote, Ultra Recall, all share very similar interfaces. There’s no need to re-invent the wheel- the developers of these well-designed applications spent decades designing their applications and interfaces. Trilium might adopt some of UI features of MyInfo such as:

- the ability to add viewable attributes, such as comments, in a column
- allow simple formatting of text in the tree to allow for colours, bold, italics, etc.

**2. Improve the tag/keyword function**
The tagging/keywording in Trilium is good, but could be considerably improved by adopting innovations used by other applications. Adding a hashtag at the beginning of each note is rather awkward. Also users should be able to add multiple words as a single keyword, a common ability in qualitative data analysis (QDA) software designed for sifting through huge volumes of text, photos, video. 

Many applications have designed outstanding keyword capabilities ie) Imatch 2020, Damation, Capture One, MyInfo, Ultra Recall, RightNote, Mybase, Digikam. One of the very best, arguably is the keywording system in the digital asset management (DAM) application “Imatch 2020”.  The Imatch developed a rather brilliant keywording system for adding keywords to images, PDF, MP3s and a host of other document formats. It is easy to search by keyword, multiple keywords, has autocomplete, recently used keywords, and a host of other convenient functions. The best tagging systems are found in qualitative data analysis software like MaxQDA, HyperResearch, Nvivo, Quirkos, AtlasTI.

**3. Allows notes to appear in multiple “virtual folders”**
Please consider adopting a “category” system similar to that in MaxQDA or Imatch 2020.  In Imatch, the “Categories” option allows users to create virtual folders.  This is a useful method in that notes can exist in in a fixed tree/hierarchical format, but at the same time, the ‘categories’ offer various other options for organization. Each note/document can be placed in multiple categories.
The explanation below should clarify the idea.

Hierarchical Tree Structure:
My city
– museum
– art gallery
– historical stories
– cultural heritage
– modern house architecture examples
– heritage houses
– industrial heritage sites
These could be regrouped into various “categories” without affecting the Hierarchical Tree Structure.
ie)
Category “nice building”
– museum
– heritage houses
– industrial heritage sites
Category “sightseeing”
– museum
– art gallery
– heritage houses
Each tagged item can appear in multiple ‘categories’.

**#4. Implement a QDA (Qualitative Data Analysis) feature- User can select portions sections of text and apply a keyword to the selected section.** 
To achieve functionality as a QDA, in Trilium, a user would be able to select portions sections of text and apply a keyword to that entire section. Ideally, each keyword/tag would appear in a margin and indicate with highlighting or coloured bars each section of text that is coded with a particular keyword. By performing a search/analysis, all sections of text coded with particular keywords can be brought together in summary reports.

This would allow Trilium to function as a simple QDA (Qualitative data analysis software). Affordable QDA software essentially does not exist, and the open source initiatives underway are extraordinarily weak and years behind Trilium. 
Having QDA functionality would strengthen Trilium’s use as a data-sorting, cataloging and analysis tool.  There are a great many esearchers and graduate students all over the world who want to use QDA software but cannot afford one of the “big names” and would be quite eager to adopt the use of Trilium as an open-source QDA. 

Please consider these suggestions for Trilium. I hope this may be of use.

**Comment by zadam** [comment]
Hi,

your post is full of ideas, but it's not really actionable. You mention a number of existing software suites, but it's not realistic for me that I download and study them.

If you want to have something added to Trilium then the best course of action is to describe one specific feature request in one issue, provide screenshots from existing applications etc.

> 3. Allows notes to appear in multiple “virtual folders”

I think this is covered by "cloning": https://github.com/zadam/trilium/wiki/Cloning-notes

**Reply by Waxwood** [reply]
Hi Zadam, thank you for your reply. I will put together some screenshots with explanations of the type of features that would introduce QDA functions and help transform Trilium into a far more powerful tool for cataloguing and analysing text data collections.

**Comment by Waxwood** [comment]
In the use-case I envisage, in any note, the user can select blocks of text and assign a keyword. When reading a webclipping in Trilium, many keywords can be assigned. Ie) the first paragraph may discuss the back-story of the article  and can be keyworded as "inductive reasoning". The second paragraph discuss interviews and can be keyworded as "ethnography" etc.  

As the user reads through their notes, they can add keywords as necessary and develop a keyword "controlled vocabulary" that they can use to assign keywords to all of their notes as necessary. 

The user can analyse the notes by performing a search showing every section of text that was assigned the keyword "ethnography". This would enable users to search for trends in the data, and linkages between ideas appearing in different texts (notes).
